bioimageio/core/build_spec/util.py

```diff
@@ -0,0 +1,21 @@
+
+
+# based on:
+# https://stackoverflow.com/questions/38491318/python-replace-keys-in-a-nested-dictionary
+def replace_values(data_dict, value_dict):
+    new_dict = {}
+    if isinstance(data_dict, list):
+        dict_value_list = list()
+        for inner_dict in data_dict:
+            dict_value_list.append(replace_values(inner_dict, value_dict))
```

Comment:
in the list case the overwrite values should probably be a dict keyed by index... ?
```suggestion
        for i, inner_dict in enumerate(data_dict):
            dict_value_list.append(replace_values(inner_dict, value_dict.get(i, {})))
```

Reply:
This doesn't make sense. The conditional here should only depend on `data_dict` and not on `value_dict`.
`value_dict` should not be a list.

Reply:
not a list, yes, but it could be a dict with index keys...
This is what I mean:
`orig_dict` = {"a": [0, 1, 2, 3]}`
`value_dict` = {"a": {3: 4}}`
`updated_dict` = {"a": [0, 1, 2, 4]}`